inc/server/class-form-server.php

```diff
@@ -7,56 +7,135 @@
 
 namespace ThemeIsle\GutenbergBlocks\Server;
 
+use Exception;
 use ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request;
 use ThemeIsle\GutenbergBlocks\Integration\Form_Data_Response;
+use ThemeIsle\GutenbergBlocks\Integration\Form_Email;
+use ThemeIsle\GutenbergBlocks\Integration\Form_Providers;
+use ThemeIsle\GutenbergBlocks\Integration\Form_Settings_Data;
 use ThemeIsle\GutenbergBlocks\Integration\Mailchimp_Integration;
 use ThemeIsle\GutenbergBlocks\Integration\Sendinblue_Integration;
+use WP_Error;
+use WP_HTTP_Response;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
 
 /**
  * Class Plugin_Card_Server
+ *
+ * @since 2.0.0
  */
 class Form_Server {
 
 	/**
 	 * The main instance var.
 	 *
 	 * @var Form_Server
+	 * @since 2.0.0
 	 */
 	public static $instance = null;
 
 	/**
 	 * Rest route namespace.
 	 *
 	 * @var Form_Server
+	 * @since 2.0.0
 	 */
 	public $namespace = 'otter/';
 
 	/**
 	 * Rest route version.
 	 *
 	 * @var Form_Server
+	 * @since 2.0.0
 	 */
 	public $version = 'v1';
 
+
 	/**
 	 * Initialize the class
+	 *
+	 * @since 2.0.0
 	 */
 	public function init() {
+		/**
+		 * Register the REST API endpoints.
+		 */
 		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+
+		/**
+		 * Add filter for validating the form data.
+		 */
+		add_filter( 'otter_form_validation', array( $this, 'check_form_conditions' ) );
+
+		/**
+		 * Add filter for captcha validation.
+		 */
+		add_filter( 'otter_form_captcha_validation', array( $this, 'check_form_captcha' ) );
```

Comment:
What was the reason for putting filters here?

Reply:
WP Forms inspired it. Show they use a hook for almost anything, including validation. If you don't see their point, I can remove them for now.